From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Heiko Becker <mail@heiko-becker.de>
Date: Thu, 20 Feb 2025 22:37:44 +0100
Subject: [PATCH] CMake: Raised required version to 3.5

CMake >= 4.0.0-rc1 removed compatibility with versions < 3.5 and errors
out with such versions passed to cmake_minimum_required(). 3.5.0 has
been released 9 years ago, so I'd assume it's available almost everywhere.

Furthermore at least 3.1 was already required, because that's the
version, which introduced CXX_STANDARD{,_REQUIRED}.

Also remove the FATAL_ERROR part, which has been ignored since 2.6. and
CMP0012 is now implicitly assumed to be NEW already.
---
 CMakeLists.txt                    |  5 +++--
 gr2fonttest/CMakeLists.txt        |  4 +++-
 src/CMakeLists.txt                | 26 +++-----------------------
 tests/bittwiddling/CMakeLists.txt |  4 +++-
 tests/json/CMakeLists.txt         |  4 +++-
 tests/sparsetest/CMakeLists.txt   |  4 +++-
 tests/utftest/CMakeLists.txt      |  4 +++-
 7 files changed, 21 insertions(+), 30 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 496712d..25db6dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,6 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
-cmake_policy(SET CMP0012 NEW)
+# SPDX-License-Identifier: MIT OR MPL-2.0 OR LGPL-2.1-or-later OR GPL-2.0-or-later
+# Copyright 2010, SIL International, All rights reserved.
+cmake_minimum_required(VERSION 3.5.0)
 include(TestBigEndian)
 find_package(PythonInterp 3.6)
 project(graphite2)
diff --git a/gr2fonttest/CMakeLists.txt b/gr2fonttest/CMakeLists.txt
index f995dd3..766ba0d 100644
--- a/gr2fonttest/CMakeLists.txt
+++ b/gr2fonttest/CMakeLists.txt
@@ -1,4 +1,6 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+# SPDX-License-Identifier: MIT OR MPL-2.0 OR LGPL-2.1-or-later OR GPL-2.0-or-later
+# Copyright 2010, SIL International, All rights reserved.
+cmake_minimum_required(VERSION 3.5.0)
 
 project(gr2fonttest)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b6ac26b..fca7184 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,27 +1,7 @@
-#    GRAPHITE2 LICENSING
-#
-#    Copyright 2010, SIL International
-#    All rights reserved.
-#
-#    This library is free software; you can redistribute it and/or modify
-#    it under the terms of the GNU Lesser General Public License as published
-#    by the Free Software Foundation; either version 2.1 of License, or
-#    (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-#    Lesser General Public License for more details.
-#
-#    You should also have received a copy of the GNU Lesser General Public
-#    License along with this library in the file named "LICENSE".
-#    If not, write to the Free Software Foundation, 51 Franklin Street,
-#    Suite 500, Boston, MA 02110-1335, USA or visit their web page on the
-#    internet at http://www.fsf.org/licenses/lgpl.html.
-
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+# SPDX-License-Identifier: MIT OR MPL-2.0 OR LGPL-2.1-or-later OR GPL-2.0-or-later
+# Copyright 2010, SIL International, All rights reserved.
+cmake_minimum_required(VERSION 3.5.0)
 project(graphite2_core)
-cmake_policy(SET CMP0012 NEW)
 INCLUDE(CheckCXXSourceCompiles)
 
 set(GRAPHITE_API_CURRENT 3)
diff --git a/tests/bittwiddling/CMakeLists.txt b/tests/bittwiddling/CMakeLists.txt
index 372b4ac..38787d1 100644
--- a/tests/bittwiddling/CMakeLists.txt
+++ b/tests/bittwiddling/CMakeLists.txt
@@ -1,4 +1,6 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+# SPDX-License-Identifier: MIT OR MPL-2.0 OR LGPL-2.1-or-later OR GPL-2.0-or-later
+# Copyright 2014, SIL International, All rights reserved.
+cmake_minimum_required(VERSION 3.5.0)
 project(bits)
 include(Graphite)
 include_directories(${graphite2_core_SOURCE_DIR})
diff --git a/tests/json/CMakeLists.txt b/tests/json/CMakeLists.txt
index ae33427..e9ea395 100644
--- a/tests/json/CMakeLists.txt
+++ b/tests/json/CMakeLists.txt
@@ -1,4 +1,6 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+# SPDX-License-Identifier: MIT OR MPL-2.0 OR LGPL-2.1-or-later OR GPL-2.0-or-later
+# Copyright 2011, SIL International, All rights reserved.
+cmake_minimum_required(VERSION 3.5.0)
 project(jsontest)
 include(Graphite)
 include_directories(${graphite2_core_SOURCE_DIR})
diff --git a/tests/sparsetest/CMakeLists.txt b/tests/sparsetest/CMakeLists.txt
index 3bfdc0c..7b940a5 100644
--- a/tests/sparsetest/CMakeLists.txt
+++ b/tests/sparsetest/CMakeLists.txt
@@ -1,4 +1,6 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+# SPDX-License-Identifier: MIT OR MPL-2.0 OR LGPL-2.1-or-later OR GPL-2.0-or-later
+# Copyright 2012, SIL International, All rights reserved.
+cmake_minimum_required(VERSION 3.5.0)
 project(sparsetest)
 include(Graphite)
 include_directories(${graphite2_core_SOURCE_DIR})
diff --git a/tests/utftest/CMakeLists.txt b/tests/utftest/CMakeLists.txt
index dc50ca4..dca0bfb 100644
--- a/tests/utftest/CMakeLists.txt
+++ b/tests/utftest/CMakeLists.txt
@@ -1,4 +1,6 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.0 FATAL_ERROR)
+# SPDX-License-Identifier: MIT OR MPL-2.0 OR LGPL-2.1-or-later OR GPL-2.0-or-later
+# Copyright 2011, SIL International, All rights reserved.
+cmake_minimum_required(VERSION 3.5.0)
 project(utftest)
 include(Graphite)
 include_directories(${graphite2_core_SOURCE_DIR})
